install.sh

```diff
@@ -35,16 +42,90 @@ fi
 sudo mkdir -p "$INTERNAL_DIR"
 sudo mkdir -p "/tmp/roller_tmp"
 
+# Function to display Go installation instructions
+GO_REQUIRED_VERSION="1.19"
+display_go_installation_instructions() {
+    echo "$EMOJI To install Go $GO_REQUIRED_VERSION, you can run the following commands:"
+    echo "  wget https://go.dev/dl/go1.19.10.${OS}-${ARCH}.tar.gz"
+    echo "  sudo rm -rf /usr/local/go && sudo tar -C /usr/local -xzf go1.19.10.${OS}-${ARCH}.tar.gz"
+}
+
 # Download and extract the tar file to a temporary directory
 echo "$EMOJI Downloading roller..."
-sudo curl -L "$TGZ_URL" --progress-bar | sudo tar -xz -C "/tmp/roller_tmp"
-
-# Assuming that the tar file contains the lib folder and the roller and rollapp_evm binaries inside the roller_bins directory.
-# Move binaries to their correct locations
-echo "$EMOJI Installing roller..."
-sudo mv "/tmp/roller_tmp/roller_bins/lib"/* "$INTERNAL_DIR"
-sudo mv "/tmp/roller_tmp/roller_bins/roller" "$ROLLER_BIN_PATH"
-sudo mv "/tmp/roller_tmp/roller_bins/rollapp_evm" "$ROLLAPP_EVM_PATH"  # move the rollapp_evm binary
+if ! sudo curl -L "$TGZ_URL" --progress-bar | sudo tar -xz -C "/tmp/roller_tmp"; then
+    # Curl failed. Fallback to clone and build.
+    echo "$EMOJI Download failed. Cloning and building manually..."
+    # Check prerequisites
+    command -v curl >/dev/null 2>&1 || { echo >&2 "$EMOJI curl is required but it's not installed. Aborting."; exit 1; }
+    command -v wget >/dev/null 2>&1 || { echo >&2 "$EMOJI wget is required but it's not installed. Aborting."; exit 1; }
+    command -v git >/dev/null 2>&1 || { echo >&2 "$EMOJI git is required but it's not installed. Aborting."; exit 1; }
+    if ! command -v go >/dev/null 2>&1; then
+        echo >&2 "$EMOJI Go is required but it's not installed. Aborting."
+        display_go_installation_instructions
+        exit 1;
+    fi
+
+    # Check go version
+    GO_VERSION=$(go version | awk -F' ' '{print substr($3, 3, 4)}')  # Extract major.minor version number (remove the 'go' prefix)
+    if [ "$GO_VERSION" != "$GO_REQUIRED_VERSION" ]; then
+        echo "$EMOJI Go version $GO_REQUIRED_VERSION is required, but you have version $GO_VERSION. Aborting."
+        display_go_installation_instructions
+        exit 1
+    fi
+
+    for i in "${!PROJECTS[@]}"; do
+        echo "$EMOJI handling ${PROJECTS[i]}..."
+        cd /tmp
+
+        # Check if binaries already exist
+        IFS=' ' read -r -a binary_array <<< "${BINARYNAME[i]}"  # convert string to array
+        binary=${binary_array[0]}  # get the first element
+        echo "$EMOJI checking for $binary..."
+        if [ -x "$(command -v "$binary")" ]; then
```

Comment:
No need for that - if the binary not exist in the roller path, we want to install it. this will cause versioning problems

Reply:
Or verify the binary version as well

Reply:
but if the binary does exist, I'm want to let the user know it's gonna be overridden